Dockerfile

```diff
@@ -0,0 +1,63 @@
+# Export Poetry Packages
+FROM ubuntu:21.10 as builder
+
+ENV PYTHONUNBUFFERED=1 \
+    PYTHONDONTWRITEBYTECODE=1 \
+    PIP_NO_CACHE_DIR=off \
+    PIP_DISABLE_PIP_VERSION_CHECK=on \
+    PIP_DEFAULT_TIMEOUT=100 \
+    POETRY_VERSION=1.1.4 \
+    POETRY_HOME="/opt/poetry" \
+    POETRY_VIRTUALENVS_IN_PROJECT=true \
+    POETRY_NO_INTERACTION=1 \
+    PYSETUP_PATH="/opt/pysetup" \
+    VENV_PATH="/opt/pysetup/.venv"
+
+# Prepend poetry and venv to path
+ENV PATH="$POETRY_HOME/bin:$VENV_PATH/bin:$PATH"
+
+# Install python 3.9
+RUN apt-get update \
+    && apt-get install -y --no-install-recommends \
+    python3.9 \
+    && apt-get clean && rm -rf /var/lib/apt/lists/* \
+    && ln -s /usr/bin/python3.9 /usr/bin/python
+
+# Install tools
+RUN apt-get update \
+    && apt-get install -y  --no-install-recommends \
+    ca-certificates \
+    curl \
+    build-essential \
+    python3-distutils \
+    && apt-get clean && rm -rf /var/lib/apt/lists/*
+
+WORKDIR /app
+COPY poetry.lock pyproject.toml /app/
+
+# Install poetry and export dependencies to requirements yaml
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
+RUN curl -sSL https://raw.githubusercontent.com/sdispater/poetry/master/get-poetry.py | python -
+
+#Set application version in pyproject.toml, use zero if not set
+ARG BUILD_NUMBER=0
+RUN poetryVersion=$(poetry version -s); buildNumber=${BUILD_NUMBER}; newVersion=$(echo $poetryVersion | sed "s/[[:digit:]]\{1,\}$/$buildNumber/"); poetry version $newVersion
+
+RUN poetry export > requirements.txt
+
+# Production image
+FROM python:3.9-slim
+
+WORKDIR /app
+COPY job_service job_service
+#To use application version in logs
+COPY --from=builder /app/pyproject.toml pyproject.toml
+COPY --from=builder /app/requirements.txt requirements.txt
+
+RUN pip install -r requirements.txt
+
+#the output is sent straight to terminal without being first buffered
+ENV PYTHONUNBUFFERED 1
+
+CMD [ "gunicorn", "job_service.app:app", "--bind", "0.0.0.0:8000", "--workers", "1"]
```

Comment:
I think this will allow only one client at a time in the service, right? It's OK since all operations on job-service don't take much time to execute.

Reply:
You are a 100% correct. As we (to my knowledge) don't have any race conditions in the code we can always increase this number if we want to utilize more cpu 👍🏻